src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden ? this.hidden = false : this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    Animal.alive.push(this);
+  }
+  bite(victim) {
+    if (!victim.hidden && !(victim instanceof Carnivore)) {
```

Comment:
I think it will be more clear

```suggestion
    if (!victim.hidden && victim instanceof Herbivore) {
```

Reply:
I agree but i wrote exactly like this because of a possibility of implementing more  Animal subclasses in the future (e.g. omnivores)